3_summarize/src/annual_thermal_metrics.R

```diff
@@ -0,0 +1,256 @@
+        max_surf_jas = max_surf_jas(date, wtr, depth),
+        mean_bot_jas = mean_bot_jas(date, wtr, depth),
+        max_bot_jas = max_bot_jas(date, wtr, depth),
+        
+        # Per month calcs return a data.frame and get "unpacked" below 
+        mean_surf_mon = mean_surf_mon(date, wtr, depth),
+        max_surf_mon = max_surf_mon(date, wtr, depth),
+        mean_bot_mon = mean_bot_mon(date, wtr, depth),
+        max_bot_mon = max_bot_mon(date, wtr, depth),
+        
+        .groups = "keep" # suppresses message about regrouping
+      ) %>% 
+      unpack(cols = c(mean_surf_mon, max_surf_mon, 
+                      mean_bot_mon, max_bot_mon))
+    
+    message(sprintf("Completed annual metrics for %s/%s", which(site_ids == id), length(site_ids)))
+    
+    return(annual_metrics)
+  }) %>% 
+    bind_rows() %>% 
+    readr::write_csv(target_name)
+  
+}
+
+# Collection of functions to calculate annual thermal metrics
+# All accept the same main data input, `date` and `wtr`.
+
+# It is assumed that these functions only get passed one
+# years worth of data.
+
+#' @param date a vector (or one column) with the dates of class `Date`
+#' @param wtr a vector (or one column) with water temperature values in deg C
+
+# Some also have the following parameters:
+#' @param depth vector (or column) with depth in meters
+
+#' @description # of days, beginning October 30 of the previous year and running 
+#' through June of the given year, that surface water is <4 degrees C
+winter_dur_0_4 <- function(date, wtr) {
+  
+}
+
+#' @description Coefficient of Variation of surface temperature from 30-60 days post ice off.
+coef_var_30_60 <- function(date, wtr) {
+  # ice dates are in 4_inputs in sciencebase
+}
+
+#' @description Coefficient of Variation of surface temperature from 0-30 days post ice off.
+coef_var_0_30 <- function(date, wtr) {
+  
+}
+
+#' @description Julian date at which stratification sets up (longest stratified period)
+stratification_onset_yday <- function(date, wtr) {
+  
+}
+
+#' @description Duration of stratified period
+stratification_duration <- function(date, wtr) {
+  
+}
+
+#' @description avg. thermocline depth in stratified period
+sthermo_depth_mean <- function(date, wtr) {
+  
+}
+
+#' @description Maximum observed surface temperature
+peak_temp <- function(date, wtr, depth) {
+  max(wtr[which(depth == 0)], na.rm = TRUE)
+}
+
+#' @description Cumulative sum of degrees > 0 for entire year
+gdd_wtr_0c <- function(date, wtr) {
+  calc_gdd(wtr, 0)
+}
+
+#' @description Cumulative sum of degrees > 5 for entire year
+gdd_wtr_5c <- function(date, wtr) {
+  calc_gdd(wtr, 5)
+ }
+
+#' @description Cumulative sum of degrees > 10 for entire year
+gdd_wtr_10c <- function(date, wtr) {
+  calc_gdd(wtr, 10)
+}
+
+#' @description water temperature 0.1m from lake bottom on day of 
+#' stratification (as defined in original stratification measure)
+bottom_temp_at_strat <- function(date, wtr) {
```

Comment:
I think this is one of the metrics that is going to rely on another metric - such as `stratification_onset_yday()`. Probably not a big deal to call that twice (once in the call above and another time in this function), but you could also pass this the date from your `summarize` call for `stratification_onset_yday`